pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -81,10 +81,25 @@ type ServiceEntryStore struct { // nolint:golint
 	seWithSelectorByNamespace map[string][]servicesWithEntry
 	refreshIndexes            *atomic.Bool
 	workloadHandlers          []func(*model.WorkloadInstance, model.Event)
+
+	processServiceEntry bool
+}
+
+type ServiceDiscoveryOption func(*ServiceEntryStore)
+
+func ProcessServiceEntry(enabled bool) ServiceDiscoveryOption {
```

Comment:
very nit: This probably makes more sense as `DisableServiceEntryProcessing()` rather than passing a bool?